docs/swagger.yaml

```diff
@@ -0,0 +1,329 @@
+          schema:
+            $ref: "#/definitions/User"
+        "500":
+          description: "Internal server error"
+    post:
+      tags:
+      - "users"
+      summary: "Add a new user to the database"
+      description: ""
+      operationId: "addUser"
+      consumes:
+      - "application/json"
+      - "application/xml"
+      produces:
+      - "application/xml"
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "body"
+        description: "User object that will be submitted to create new user"
+        required: true
+        schema:
+          $ref: "#/definitions/User"
+      responses:
+        "200":
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/User"
+        "500":
+          description: "Internal server error"
+
+  /users/{user_id}:
+    get:
+      tags:
+      - "users/{user_id}"
+      summary: "Get a single user by user_id"
+      description: ""
+      operationId: "getUserByName"
+      produces:
+      - "application/xml"
+      - "application/json"
+      parameters:
+      - name: "user_id"
+        in: "path"
+        description: "The name that needs to be fetched. Use user1 for testing. "
+        required: true
+        type: "string"
+      responses:
+        "200":
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/User"
+        "404":
+          description: "User not found"
+        "500":
+          description: "Internal Server Error"
+    put:
+      tags:
+      - "users/{user_id}"
+      summary: "Updated single user by user_id"
+      description: "This can only be done by the logged in user."
+      operationId: "updateUser"
+      produces:
+      - "application/xml"
+      - "application/json"
+      parameters:
+      - name: "user_id"
+        in: "path"
+        description: "name that need to be updated"
+        required: true
+        type: "string"
+      - in: "body"
+        name: "body"
+        description: "Updated user object"
+        required: true
+        schema:
+          $ref: "#/definitions/User"
+      responses:
+        "200":
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/User"
+        "404":
+          description: "User not found"
+        "500":
+          description: "Internal Server Error"
+    delete:
+      tags:
+      - "users/{user_id}"
+      summary: "Delete a single user by user_id"
+      description: "This can only be done by the logged in user."
+      operationId: "deleteUser"
+      produces:
+      - "application/xml"
+      - "application/json"
+      parameters:
+      - name: "user_id"
+        in: "path"
+        description: "The name that needs to be deleted"
+        required: true
+        type: "string"
+      responses:
+        "200":
+          description: "User Deleted"
+        "400":
+          description: "Delete Failed"
```

Comment:
Maybe add a "404": if the user was not found? but 400 would include it too.

Reply:
Also I think the function for "users/{user_id}" includes a patch method and would need to be included ( if a user updates something)

Reply:
It does have a patch!